drivers/soundwire/bus.c

```diff
@@ -897,6 +901,10 @@ static int sdw_bus_wait_for_clk_prep_deprep(struct sdw_bus *bus, u16 dev_num)
 	do {
 		val = sdw_bread_no_pm(bus, dev_num, SDW_SCP_STAT) &
 			SDW_SCP_STAT_CLK_STP_NF;
+		if (val < 0) {
```

Comment:
this cannot work: `val` is and-ed with `BIT(0)` above

Reply:
nice catch, thanks @lyakh. Fixed now.